java/src/main/java/ai/rapids/cudf/Rmm.java

```diff
@@ -56,18 +95,62 @@ public static void initialize(int allocationMode, boolean enableLogging, long po
    */
   public static void initialize(int allocationMode, boolean enableLogging, long poolSize, int gpuId)
       throws RmmException {
-    if (defaultInitialized) {
-      synchronized(Rmm.class) {
-        if (defaultInitialized) {
-          shutdown();
-          defaultInitialized = false;
-        }
+    LogConf lc = null;
+    if (enableLogging) {
+      String f = System.getenv("RMM_LOG_FILE");
+      if (f != null) {
+        lc = logTo(new File(f));
+      } else {
+        lc = logToStderr();
       }
     }
-    initializeInternal(allocationMode, enableLogging, poolSize);
-    if (gpuId >= 0) {
-      MemoryCleaner.setDefaultGpu(gpuId);
+    initialize(allocationMode, lc, poolSize, gpuId);
+  }
+
+  /**
+   * Initialize memory manager state and storage.
+   * @param allocationMode Allocation strategy to use. Bit set using
+   *                       {@link RmmAllocationMode#CUDA_DEFAULT},
+   *                       {@link RmmAllocationMode#POOL} and
+   *                       {@link RmmAllocationMode#CUDA_MANAGED_MEMORY}
+   * @param logConf        How to do logging or null if you don't want to
+   * @param poolSize       The initial pool size in bytes
+   * @param gpuId          The GPU that RMM should use.  You are still responsible for
+   *                       setting this GPU yourself on each thread before calling RMM
+   *                       this just sets it for any internal threads used.
+   * @throws IllegalStateException if RMM has already been initialized
+   */
+  public static synchronized void initialize(int allocationMode, LogConf logConf, long poolSize, int gpuId)
+      throws RmmException {
+    if (initialized) {
+      throw new IllegalStateException("RMM is already initialized");
     }
+    int logTo = 0; // NONE
+    String path = null;
+    if (logConf != null) {
+      if (logConf.location != null) {
+        logTo = 1;
```

Comment:
It would be cleaner and clearer if these magic values were a private enum (with their corresponding native values) and used directly in the `LogConf` class rather than separate double-booleans to track stdout vs. stderr (and the invalid state of stdout _and_ stderr).